src/SpeedCalc.cc

```diff
@@ -84,6 +84,29 @@ int SpeedCalc::calculateSpeed()
   return speed;
 }
 
+int SpeedCalc::calculateNewestSpeed(int seconds)
+{
+  const auto& now = global::wallclock();
+  removeStaleTimeSlot(now);
+
+  int64_t bytesCount(0);
+  auto it=timeSlots_.rbegin();
+  while (it != timeSlots_.rend()) {
+    if (it->first.difference(now) > seconds*1_s) {
+      break;
+    }
+    bytesCount += (*it++).second;
+  }
+  if (it == timeSlots_.rbegin()) return 0;
```

Comment:
I think we should return the sum of received bytes including first slot.  If all slots are within given seconds, all slots should be considered.